HPXMLtoOpenStudio/resources/hpxml_defaults.rb

```diff
@@ -1525,6 +1526,33 @@ def self.apply_generators(hpxml)
     end
   end
 
+  def self.apply_batteries(hpxml)
+    default_values = Battery.get_battery_default_values()
+    hpxml.batteries.each do |battery|
+      if battery.location.nil?
+        battery.location = default_values[:location]
+        battery.location_isdefaulted = true
+      end
+      if battery.lifetime_model.nil?
+        battery.lifetime_model = default_values[:lifetime_model]
+        battery.lifetime_model_isdefaulted = true
+      end
+      if battery.capacity.nil?
+        if battery.voltage.nil?
+          capacity = Battery.get_Ah_from_kWh(default_values[:capacity], default_values[:voltage])
+        else
+          capacity = Battery.get_Ah_from_kWh(default_values[:capacity], battery.voltage)
+        end
```

Comment:
THis could be simplified to just `capacity = Battery.get_Ah_from_kWh(default_values[:capacity], battery.voltage)` if you process a nil battery voltage before you process a nil battery capacity.